x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/AllocationRoutedStep.java

```diff
@@ -71,7 +71,7 @@ public Result isConditionMet(Index index, ClusterState clusterState) {
                 boolean canRemainOnCurrentNode = ALLOCATION_DECIDERS
                         .canRemain(shardRouting, clusterState.getRoutingNodes().node(currentNodeId), allocation)
                         .type() == Decision.Type.YES;
-                if (canRemainOnCurrentNode == false) {
+                if (canRemainOnCurrentNode == false || shardRouting.started() == false || shardRouting.relocating()) {
```

Comment:
Can started() return true if its relocating?

Reply:
No, the state would be either STARTED or RELOCATING, so I suppose the the started check is needed. I'll remove the relocating check